### PR TITLE
:bug: (go/v4) update gitignore template comments and fix some nit typo issues

### PR DESCRIPTION
**Pull request by mateusoliveira43**
## Description
Update gitignore template  for golang/v4 plugin (v2 and v3 are deprecated and will be removed soon) to be up to date with https://github.com/github/gitignore/blob/main/Go.gitignore

## Motivation
Use the default template of GitHub

## Related issues
No related issues

### CODE REVIEW

File: pkg/plugins/golang/v2/scaffolds/internal/templates/gitignore.go [hunk 1]
```diff
@@ -47,14 +47,16 @@ const gitignoreTemplate = `
 *.dylib
 bin
```
Comment (camilamacedo86) [inline pkg/plugins/golang/v2/scaffolds/internal/templates/gitignore.go:49]: The plugins go/v2 and go/v3 are deprecated for too long and will be removed in the first quarter
Therefore, could you please revert the changes on those?

**Review by camilamacedo86 — COMMENTED** [review]
Thank you for your contribution 🥇 

Note that the plugins go/v2, go/v3 are deprecated for a long period so we cannot change things on them. Therefore, could you please revert the changes made on those?

Also, see that you need to sign the CLA for we are able to get this PR merged, can you please check it out?

File: pkg/plugins/golang/v2/scaffolds/internal/templates/gitignore.go (hunk 1 above)
Comment (mateusoliveira43) [reply pkg/plugins/golang/v2/scaffolds/internal/templates/gitignore.go:49]: sure, I will try to this in the next days. I was in doubt about CLA option, once I figure it out, I update it and the PR

Comment (camilamacedo86) [reply pkg/plugins/golang/v2/scaffolds/internal/templates/gitignore.go:49]: Hi @mateusoliveira43 

Thank you. See here the steps to sign the CLA, you just need to register yourself: https://github.com/kubernetes-sigs/kubebuilder/pull/3717#issuecomment-1872282156 (less then 2 minutes)

Also, more one nit. Since it will change something for the endusers the right emoji is 🐛 and not 🌱 . More info can be found in the Contributing guide.

Comment (mateusoliveira43) [reply pkg/plugins/golang/v2/scaffolds/internal/templates/gitignore.go:49]: I will update the emoji as well

My question with CLA is if I am Individual Contributor or Corporate Contributor. Once I get that checked, I will update the PR

Comment (mateusoliveira43) [reply pkg/plugins/golang/v2/scaffolds/internal/templates/gitignore.go:49]: sorry the delay

Now that the yer change, `make generate` produces a lot of changes. Need to have a separate PR for just the year change. Do you want me to create this PR?

**Comment by mateusoliveira43** [comment]
I wait https://github.com/kubernetes-sigs/kubebuilder/pull/3719 get merged before finishing this one, to rebase on top of it 

Is the requirement of squashing the commits in contribute guidelines?

**Comment by camilamacedo86** [comment]
Hi @mateusoliveira43,

Could you please:
- Rebase the PR with master
- Squash the commits 

So that we can move forward here?

**Comment by mateusoliveira43** [comment]
@camilamacedo86 done

**Review by camilamacedo86 — APPROVED** [review]
/lgtm

/approved

/ok-to-test
